core/digital-twin-lib/src/dispatchers/description.ts

```diff
@@ -17,22 +17,35 @@
   the following URL: https://evan.network/license/
 */
 
-import * as dappBrowser from '@evan.network/ui-dapp-browser';
+import { getDomainName, ipfs, } from '@evan.network/ui-dapp-browser';
 import { Dispatcher, DispatcherInstance } from '@evan.network/ui';
+import { Ipfs } from '@evan.network/api-blockchain-core';
 
 const dispatcher = new Dispatcher(
-  `lib.digital-twin.${ dappBrowser.getDomainName() }`,
+  `lib.digital-twin.${ getDomainName() }`,
   'descriptionDispatcher',
   40 * 1000,
   '_digital-twin-lib.dispatchers.description'
 );
 
 dispatcher
+  // upload image into ipfs
+  .step(async (instance: DispatcherInstance, { description, } ) => {
+    if (description.imgSquare && typeof description.imgSquare !== 'string') {
+      const imageBuffer = Buffer.from(description.imgSquare.file);
+
+      // upload file, build correct ipfs url and build the img url
+      const uploaded = await instance.runtime.dfs.add(description.imgSquare.name, imageBuffer);
+      const ipfsHash = Ipfs.bytes32ToIpfsHash(uploaded);
+      const { host, port, protocol } = ipfs.ipfsConfig;
+      description.imgSquare = `${ protocol }://${ host }:${port}/ipfs/${ ipfsHash }`;
```

Comment:
is it clever to store the fully url in the contract? In case the config changes at any point in the future the links wouldn't work anymore right?

Reply:
a helper function hashToUrl could do this also dynamically on rendering.

Reply:
on the other hand we could also offer the possibility to link external resources in the future.

Reply:
Has Advantages and Disadvantages. I would prefer to only store images that can be directly opened without adjustments.